common/modules/farm/farm_base_impl/src/base_traits_impl.rs

```diff
@@ -0,0 +1,177 @@
+elrond_wasm::imports!();
+
+use common_structs::{FarmToken, FarmTokenAttributes};
+use contexts::storage_cache::StorageCache;
+use core::any::TypeId;
+use elrond_wasm::elrond_codec::TopEncode;
+use fixed_supply_token::FixedSupplyToken;
+use mergeable::Mergeable;
+
+pub trait AllBaseFarmImplTraits =
+    rewards::RewardsModule
+        + config::ConfigModule
+        + farm_token::FarmTokenModule
+        + permissions_module::PermissionsModule
+        + pausable::PausableModule
+        + events::EventsModule
+        + elrond_wasm_modules::default_issue_callbacks::DefaultIssueCallbacksModule;
+
+pub trait FarmContract {
+    type FarmSc: AllBaseFarmImplTraits;
+    // type Api: VMApi = <Self::FarmSc as ContractBase>::Api;
+
+    type AttributesType: 'static
+        + Clone
+        + TopEncode
+        + TopDecode
+        + NestedEncode
+        + NestedDecode
+        + Mergeable<<Self::FarmSc as ContractBase>::Api>
+        + FixedSupplyToken<<Self::FarmSc as ContractBase>::Api>
+        + FarmToken<<Self::FarmSc as ContractBase>::Api> =
+        FarmTokenAttributes<<Self::FarmSc as ContractBase>::Api>;
+
+    #[inline]
+    fn mint_rewards(
```

Comment:
we actually do not have any mint here - we do not want mint anymore for new contracts as they give out LKMEX created by simple lock energy.

Reply:
Maybe we update this in a future PR, one that includes the entire flow, with both farm & simple-lock-energy contract updates.

Reply:
I think we should first focus on the FARM we need, that one does not make mint and ask simple lock with energy contract to create and send tokens to the user. Let's do that first. Later we can come back and make something else.

Reply:
Mint function can be easily overwritten with an empty function, so no actual token minting is done. Will do this when the optimization for "no needless mint/burn" PR.